pandas/tests/plotting/test_misc.py

```diff
@@ -549,3 +550,13 @@ def test_externally_shared_axes(self):
         assert not plots[0][0].xaxis.get_label().get_visible()
         assert plots[0][1].xaxis.get_label().get_visible()
         assert not plots[0][2].xaxis.get_label().get_visible()
+
+    def test_plot_bar_axis_units_timestamp_conversion(self):
+        # GH 38736
+        # Ensure string x-axis from the second plot will not be converted to datetime
+        # due to axis data from first plot
+        DataFrame(
```

Comment:
Could you add some sort of assertion to these calls instead of just being a smoke test?

Reply:
Added the `_check_plot_works` function to each plot. 

I also want to assert the datatype of the axis of the plots (ie. datetime and string), but I am a bit lost on how I can achieve that.